Core/include/Acts/EventData/ParameterSet.hpp

```diff
@@ -578,29 +402,37 @@ class ParameterSet {
    * @param values vector with parameter values to be checked and corrected if
    * necessary
    */
-  static void correctValues(ParameterVector& values) {
-    detail::value_corrector<params...>::result(values);
+  static void correctValues(ParametersVector& values) {
+    detail::ValueCorrector<parameter_indices_t, params...>::run(values);
   }
 
  private:
-  ParameterVector m_vValues{
-      ParameterVector::Zero()};  ///< column vector containing
-                                 ///< values of local parameters
-  std::optional<CovarianceMatrix> m_optCovariance{
-      std::nullopt};  ///< an optional covariance matrix
-
-  static const Projection sProjector;  ///< matrix to project full parameter
-                                       /// vector onto local parameter space
+  /// column vector containing values of local parameters.
+  ParametersVector m_vValues{ParametersVector::Zero()};
+  /// optional covariance matrix.
+  std::optional<CovarianceMatrix> m_optCovariance{std::nullopt};
+
+  /// matrix to project full parameter vector onto local parameter space.
+  static const ProjectionMatrix sProjector;
+
+  /// Compare two parameter sets for value equality.
+  friend constexpr bool operator==(const ParameterSet& lhs,
+                                   const ParameterSet& rhs) {
+    return (lhs.m_vValues == rhs.m_vValues) and
+           (lhs.m_optCovariance == rhs.m_optCovariance);
+  }
+  /// Compare two parameters for value inequality.
+  friend constexpr bool operator!=(const ParameterSet& lhs,
+                                   const ParameterSet& rhs) {
+    return !(lhs == rhs);
+  }
 };
 
-// initialize static class members
-template <typename parameter_indices_t, parameter_indices_t... params>
-constexpr unsigned int
-    ParameterSet<parameter_indices_t, params...>::kNumberOfParameters;
-
-template <typename parameter_indices_t, parameter_indices_t... params>
-const typename ParameterSet<parameter_indices_t, params...>::Projection
-    ParameterSet<parameter_indices_t, params...>::sProjector =
+template <typename parameter_indices_t, parameter_indices_t... kParameters>
```

Comment:
Just about the name: does this naming not imply this is a constant, somehow? (it isn't)

Reply:
Yes, because it is closest to a constant value. It represents a value, it is known at compile time and does not change. It is a bit peculiar since it references not just one constant but all constants in the pack. However, within an expression it is always used as if it where a single constant. The unpacking usually unfolds the full expression. That is why I used the constant naming convention here.

I thought that we had added this to the naming guidelines, but it only discusses template types.

Reply:
But it's not at all constant. It's variable depending on what you plug in.

Reply:
Depends on which level you are looking at. If you look at the expression level, it definitely is a constant (non-changing, non-assignable value). If you look at it from a meta level it is not. I am not sure which way is the right way to address this.

What are your suggestions for naming instead?

Reply:
By that logic, in 

```cpp
struct A {};
void foo(const A& a);
```

`a` would be a *constant* here, which it isn't in the sense of the word we're talking about here.

My suggestion: `parameters` of `params` (as it was)?

Reply:
I disagree. In your example, `a` can still change at runtime (or better the underlying value) and you can not reason about its value at compile time. Both for
```
constexpr int kX = 2;
```
and for instantiations of
```
template<int... kX>
struct { ... };
```
the value of `kX` can not change at runtime. You can reason about it already at compile time. My initial explanation (and choice of words) was probably not clear enough.

I am not 100% happy with `parameters`, i.e. naming it as a regular variable. With this you end up assigning variables to constants, e.g.
```
template<size_t sizeOfSomething>
struct X {
  constexpr size_t kSizeOfSomething = sizeOfSomething;
}
```
which is inconsistent.

I would prefer not to introduce an third naming option for variable-like objects. So it is either `kParameters` or `parameters`.

Reply:
Since we based some of our guidelines on the Google styleguide: they suggests [constant naming](https://google.github.io/styleguide/cppguide.html#Constant_Names) for variables whose value is fixed for the duration of the program. It does not value template parameters explicitly, but it would fall under that definition.

Reply:
Ok.